authors/apps/articles/views.py

```diff
@@ -99,3 +99,140 @@ def retrieve(self, request, slug):
         serializer = self.detailSerializer()
         serializer.instance = article_instance
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+class RetrieveArticleWithReadingTime(generics.RetrieveAPIView):
+
+    def get_queryset(self):
+
+        queryset = Article.objects.all()
+
+        return queryset
+
+    def retrieve(self, *args):
+
+        articles = self.get_queryset()
+
+        article_body = []
+
+        title = {}
+
+        description = {}
+
+        body = {}
+
+        time = {}
+
+        created_at = {}
+
+        WPM = 200
+
+        for article in articles:
+
+            the_full_sentence = "{} {}".format(article.title, article.body)
+
+            words = the_full_sentence.split()
+
+            number_of_words = len(words)
+
+            minutes = round(number_of_words/WPM,0)
+
+            if int(minutes) < 1:
```

Comment:
Also, doing  this sort of operation every time articles are being retrieved  can be expensive and hence inefficient, so instead of doing it every time articles are being fetched, you could say store the article's read time in the database once since it doesn't change.

The best time to do this would be when the article is being created or updated

Reply:
This makes a lot of sense. Thanks Roland